src/plugins/data_views/common/utils.ts

```diff
@@ -19,15 +20,12 @@ import { DATA_VIEW_SAVED_OBJECT_TYPE } from './constants';
  * @returns {SavedObject|undefined}
  */
 export async function findByName(client: SavedObjectsClientCommon, name: string) {
-  if (name) {
-    const savedObjects = await client.find<DataViewSavedObjectAttrs>({
-      type: DATA_VIEW_SAVED_OBJECT_TYPE,
-      perPage: 10,
-      search: `"${name}"`,
-      searchFields: ['name'],
-      fields: ['name'],
-    });
+  const savedObjects = await client.find<{ name: DataViewSavedObjectAttrs['name'] }>({
+    type: DATA_VIEW_SAVED_OBJECT_TYPE,
+    perPage: 10000,
+    search: `"${escapeKuery(name)}"`,
+    searchFields: ['name.keyword'],
+  });
 
-    return savedObjects ? savedObjects[0] : undefined;
-  }
+  return savedObjects ? savedObjects[0] : undefined;
```

Comment:
I think nothing is guarding that `name` is an empty string, which would trigger a useless call to `client.find`.

Can we leave in the `if (name) {}` or simplify it to something like:
```
  const savedObjects = name && await client.find<{ name: DataViewSavedObjectAttrs['name'] }>({
    type: DATA_VIEW_SAVED_OBJECT_TYPE,
    perPage: 10000,
    search: `"${escapeKuery(name)}"`,
    searchFields: ['name.keyword'],
  });
```